feat(opensearch): add support for latest amazon opensearch service 2.3

agusti-t opened the pull request:
fixes https://github.com/aws/aws-cdk/issues/22942 
Description:
AWS Launched the support for Amazon OpenSearch Service 2.3 as per the announcement here: https://aws.amazon.com/about-aws/whats-new/2022/11/amazon-opensearch-service-supports-opensearch-version-2-3/

AWS CDK does not support the latest version yet, and can not create an OpenSearch cluster with CDK using 2.3.

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Unconventional Dependencies:

* [ ] This PR adds new unconventional dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-new-unconventional-dependencies)

### New Features

* [ ] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [ ] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b8ddf4c811fad42326ce80d871bf1d109252b589
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=2fe16709-3c4a-4683-9ac5-24510cd8b4c1%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0edafdf28cef26b52776f0f397041afa8001aeb9
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=be637b97-b058-4023-8c38-57ad1e2e1247%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 4346e92d22945adc76bc7f77b2e9bbf5cb4c58f0
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=08c317bb-1bf9-46e8-9810-cc1ae8fcc81c%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: eecb475d3ec33a2f14e8ce369c39171906446f97
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b2336cfd-9d13-4186-815d-aca98298a0df%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra reviewed (dismissed):
Thank you for your contribution! 

The minimal changes in the snapshot of the integration test usually indicates that the tests weren't actually run. Please revert the change  in the snapshot and run `yarn integ --update-on-failed` to validate this change and to produce the new snapshots. I also notice that the integ test isn't using the new [IntegTest](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md) construct. Please update that.

aws-cdk-automation reviewed (dismissed):
The pull request linter has failed. See the aws-cdk-automation comment below for failure reasons. If you believe this pull request should receive an exemption, please comment and provide a justification.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.

agusti-t commented:
> Thank you for your contribution!
> 
> The minimal changes in the snapshot of the integration test usually indicates that the tests weren't actually run. Please revert the change in the snapshot and run `yarn integ --update-on-failed` to validate this change and to produce the new snapshots. I also notice that the integ test isn't using the new [IntegTest](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md) construct. Please update that.

Thanks for the review, I tried using the new `IntegTest` and run the tests as you said and run into this problem:

```
➜  aws-cdk git:(OpenSearch_2_3) ✗ yarn integ-runner --directory packages/@aws-cdk/aws-appsync --update-on-failed
yarn run v1.22.19
$ /Users/agusti/src/aws-cdk/node_modules/.bin/integ-runner --directory packages/@aws-cdk/aws-appsync --update-on-failed

Verifying integration test snapshots...

  UNCHANGED  integ.appsync-none 0.915s
  UNCHANGED  integ.log-retention 1.066s
  UNCHANGED  integ.api-import 1.092s
  UNCHANGED  integ.appsync-lambda 1.082s
  CHANGED    integ.graphql-opensearch 1.214s
      appsync-opensearch exists in snapshot, but not in actual
  CHANGED    integ.graphql-opensearch 1.214s
      OpensSearch23Stack does not exist in snapshot, but does in actual
  UNCHANGED  integ.auth-apikey 1.92s
  UNCHANGED  integ.graphql-schema 1.919s
  UNCHANGED  integ.graphql-iam 1.948s
  UNCHANGED  integ.graphql-elasticsearch 1.972s
  UNCHANGED  integ.graphql 2.045s

Snapshot Results:

Tests:    1 failed, 10 total

Running integration tests for failed tests...

Running in parallel across regions: us-east-1, us-east-2, us-west-2
Running test /Users/agusti/src/aws-cdk/packages/@aws-cdk/aws-appsync/test/integ.graphql-opensearch.js in us-east-1
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
!!                                                                                  !!
!!  This software has not been tested with node v19.0.1.                            !!
!!  You may to encounter runtime issues, and should switch to a supported release.  !!
!!                                                                                  !!
!!  This software is currently running on node v19.0.1.                             !!
!!  As of the current release of this software, supported node releases are:        !!
!!  - ^18.0.0 (Planned end-of-life: 2025-04-30)                                     !!
!!  - ^16.3.0 (Planned end-of-life: 2023-09-11)                                     !!
!!  - ^14.6.0 (Planned end-of-life: 2023-04-30)                                     !!
!!                                                                                  !!
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!

✨  Synthesis time: 0.3s

OpensSearch23Stack: building assets...

[0%] start: Building 7725c14cf18390409c0bad4e2d2037a44b300aac680ce62082d410f076e7f331:current_account-current_region
[100%] success: Built 7725c14cf18390409c0bad4e2d2037a44b300aac680ce62082d410f076e7f331:current_account-current_region

OpensSearch23Stack: assets built

lambda-api-latebound-deploymentstage/DefaultTest/DeployAssert: building assets...

[0%] start: Building 21fbb51d7b23f6a6c262b46a9caee79d744a3ac019fd45422d988b96d44b2a22:current_account-current_region
[100%] success: Built 21fbb51d7b23f6a6c262b46a9caee79d744a3ac019fd45422d988b96d44b2a22:current_account-current_region

lambda-api-latebound-deploymentstage/DefaultTest/DeployAssert: assets built

OpensSearch23Stack
OpensSearch23Stack: deploying...
[0%] start: Publishing 7725c14cf18390409c0bad4e2d2037a44b300aac680ce62082d410f076e7f331:current_account-current_region
[100%] success: Published 7725c14cf18390409c0bad4e2d2037a44b300aac680ce62082d410f076e7f331:current_account-current_region
OpensSearch23Stack: creating CloudFormation changeset...
OpensSearch23Stack | 0/10 | 12:38:57 AM | REVIEW_IN_PROGRESS   | AWS::CloudFormation::Stack     | OpensSearch23Stack User Initiated
OpensSearch23Stack | 0/10 | 12:39:04 AM | CREATE_IN_PROGRESS   | AWS::CloudFormation::Stack     | OpensSearch23Stack User Initiated
OpensSearch23Stack | 0/10 | 12:39:09 AM | CREATE_IN_PROGRESS   | AWS::AppSync::GraphQLApi       | api (apiC8550315)
OpensSearch23Stack | 0/10 | 12:39:09 AM | CREATE_IN_PROGRESS   | AWS::IAM::User                 | User (User00B015A1)
OpensSearch23Stack | 0/10 | 12:39:10 AM | CREATE_IN_PROGRESS   | AWS::IAM::Role                 | api/ds/ServiceRole (apidsServiceRoleBDB08107)
OpensSearch23Stack | 0/10 | 12:39:10 AM | CREATE_IN_PROGRESS   | AWS::IAM::User                 | User (User00B015A1) Resource creation Initiated
OpensSearch23Stack | 0/10 | 12:39:10 AM | CREATE_FAILED        | AWS::IAM::Role                 | api/ds/ServiceRole (apidsServiceRoleBDB08107) Invalid principal in policy: "SERVICE":"appsync" (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: d3941825-78dd-4440-9132-0254299af33f; Proxy: null)
OpensSearch23Stack | 0/10 | 12:39:13 AM | CREATE_IN_PROGRESS   | AWS::AppSync::GraphQLApi       | api (apiC8550315) Resource creation Initiated
OpensSearch23Stack | 1/10 | 12:39:13 AM | CREATE_COMPLETE      | AWS::AppSync::GraphQLApi       | api (apiC8550315)
OpensSearch23Stack | 1/10 | 12:39:16 AM | CREATE_IN_PROGRESS   | AWS::AppSync::GraphQLSchema    | api/Schema (apiSchema0EA92056)
OpensSearch23Stack | 1/10 | 12:39:18 AM | CREATE_IN_PROGRESS   | AWS::AppSync::GraphQLSchema    | api/Schema (apiSchema0EA92056) Resource creation Initiated
OpensSearch23Stack | 2/10 | 12:39:46 AM | CREATE_COMPLETE      | AWS::IAM::User                 | User (User00B015A1)
OpensSearch23Stack | 2/10 | 12:39:47 AM | CREATE_IN_PROGRESS   | AWS::OpenSearchService::Domain | Domain (Domain66AC69E0)
OpensSearch23Stack | 2/10 | 12:39:50 AM | CREATE_IN_PROGRESS   | AWS::OpenSearchService::Domain | Domain (Domain66AC69E0) Resource creation Initiated
OpensSearch23Stack | 3/10 | 12:40:19 AM | CREATE_COMPLETE      | AWS::AppSync::GraphQLSchema    | api/Schema (apiSchema0EA92056)
OpensSearch23Stack | 3/10 | 12:40:22 AM | CREATE_IN_PROGRESS   | AWS::AppSync::ApiKey           | api/DefaultApiKey (apiDefaultApiKey6AB8D7C4)
OpensSearch23Stack | 3/10 | 12:40:24 AM | CREATE_IN_PROGRESS   | AWS::AppSync::ApiKey           | api/DefaultApiKey (apiDefaultApiKey6AB8D7C4) Resource creation Initiated
OpensSearch23Stack | 4/10 | 12:40:24 AM | CREATE_COMPLETE      | AWS::AppSync::ApiKey           | api/DefaultApiKey (apiDefaultApiKey6AB8D7C4)
OpensSearch23Stack | 5/10 | 12:56:02 AM | CREATE_COMPLETE      | AWS::OpenSearchService::Domain | Domain (Domain66AC69E0)
OpensSearch23Stack | 5/10 | 12:56:03 AM | CREATE_FAILED        | AWS::CloudFormation::Stack     | OpensSearch23Stack The following resource(s) failed to create: [apidsServiceRoleBDB08107].

Failed resources:
OpensSearch23Stack | 12:39:10 AM | CREATE_FAILED        | AWS::IAM::Role                 | api/ds/ServiceRole (apidsServiceRoleBDB08107) Invalid principal in policy: "SERVICE":"appsync" (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: d3941825-78dd-4440-9132-0254299af33f; Proxy: null)

 ❌  OpensSearch23Stack failed: Error: The stack named OpensSearch23Stack failed to deploy: CREATE_FAILED (The following resource(s) failed to create: [apidsServiceRoleBDB08107]. ): Invalid principal in policy: "SERVICE":"appsync" (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: d3941825-78dd-4440-9132-0254299af33f; Proxy: null)
    at FullCloudFormationDeployment.monitorDeployment (/Users/agusti/src/aws-cdk/packages/aws-cdk/lib/api/deploy-stack.ts:505:13)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at deployStack (/Users/agusti/src/aws-cdk/packages/aws-cdk/lib/cdk-toolkit.ts:264:24)
    at /Users/agusti/src/aws-cdk/packages/aws-cdk/lib/deploy.ts:39:11
    at run (/Users/agusti/src/aws-cdk/node_modules/p-queue/dist/index.js:163:29)

 ❌ Deployment failed: Error: Stack Deployments Failed: Error: The stack named OpensSearch23Stack failed to deploy: CREATE_FAILED (The following resource(s) failed to create: [apidsServiceRoleBDB08107]. ): Invalid principal in policy: "SERVICE":"appsync" (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: d3941825-78dd-4440-9132-0254299af33f; Proxy: null)
    at Object.exports.deployStacks (/Users/agusti/src/aws-cdk/packages/aws-cdk/lib/deploy.ts:61:11)
    at CdkToolkit.deploy (/Users/agusti/src/aws-cdk/packages/aws-cdk/lib/cdk-toolkit.ts:338:7)
    at initCommandLine (/Users/agusti/src/aws-cdk/packages/aws-cdk/lib/cli.ts:364:12)

Stack Deployments Failed: Error: The stack named OpensSearch23Stack failed to deploy: CREATE_FAILED (The following resource(s) failed to create: [apidsServiceRoleBDB08107]. ): Invalid principal in policy: "SERVICE":"appsync" (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: d3941825-78dd-4440-9132-0254299af33f; Proxy: null)
  FAILED     integ.graphql-opensearch-lambda-api-latebound-deploymentstage/DefaultTest (undefined/us-east-1) 2389.226s
      Integration test failed: Error: Command exited with status 1

Test Results:

Tests:    1 failed, 1 total
Error: Some integration tests failed!
    at main (/Users/agusti/src/aws-cdk/packages/@aws-cdk/integ-runner/lib/cli.js:114:23)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Any idea what it could be? I tried looking it up without being able to find anything.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 534ccb7e368f6be6a4d633eee2e35eaa39c200e0
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=1221803d-9f74-45d9-89ca-b1253fb3c036%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d2313a4ea7c609fad7fee1073b03a57682a72353
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f6a3436f-ded6-40c4-92a4-39be6b01d0ae%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d2313a4ea7c609fad7fee1073b03a57682a72353
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6aa09bef-ceb1-4ce8-8cd7-b361e13eb204%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 99d714193b7d86d263971739f9b843c8ca78b540
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=b23ddd0e-e5f3-4cbe-8801-444d3f22089f%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 983198f5fc9ee3748923792272c620ca5f8da535
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=69d4228f-8511-4319-b3d3-cfd5378a2736%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra commented:
@Mergifyio update

aws-cdk-automation commented:
The pull request linter fails with the following errors:

	❌ Features must contain a change to an integration test file and the resulting snapshot.

PRs must pass status checks before we can provide a meaningful review.

TheRealAmazonKendra reviewed (dismissed):
Please see my comments inline.

We had a bug that was causing your test failures. I updated your branch to be up-to-date with `main` so the fix is in there now. You should now be able to run `yarn integ --update-on-failed` and get the correct output. Note that you may need to clean the generated files from the build because the test runner is expecting to see certain files that don't exist due to the update in stack name and using the new `IntegTest` construct. 

Please let me know if you have any questions about this.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9753d8a76af94f55993f065c12f30fca9b4877f9
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=882effec-e131-4df7-82b6-36d5f652c1f9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

agusti-t commented:
@TheRealAmazonKendra Trying again, it took me a while to be able to run the tests with `yarn integ --update-on-failed`, it kept complaining about not finding the `integ` cmd, TBH I don't know how/why it started working. I had to keep trying with `yarn integ-runner --update-on-failed` and that always failed. It could also be that I rebased `main` and got the bug (suspect [this](https://github.com/aws/aws-cdk/commit/0bfce8965ee50ab79054e6f5a4c6bbecb0955e19) one) fix in, don't understand why I didn't get it earlier.

Anyway, it worked on my machine now and I committed everything, but I ask you to please check commits ([1](https://github.com/aws/aws-cdk/pull/22943/commits/544911ce7726f4a640382bbd0024ac09b9082cae), [2](https://github.com/aws/aws-cdk/pull/22943/commits/56ec81f4878d86f234cca0f122e4824faa2fde2a)) because I am not sure if these files should be committed. Of course, the rest of the commits should be checked too.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: e1729c55548da77f5d0b0e19a2e0b58ac7df625c
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=0ae3c409-41d0-4f61-b91a-52b3169c701e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: fe52135c38d3281bc7da43ef4f5b5fff7aa40d47
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=6b432a45-26fd-4bc5-84ef-7252e151afb7%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 7e63111b0f08126d79bed5731baad38597646c4a
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=a303bc69-074f-4321-aa87-f8e7ce4791ba%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 3eede783b9e2e58fb9db99f7a02c9c22fdd5f129
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=58ea2f8c-2321-44a6-8db3-91c4c1b8485d%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

Naumel commented:
@Mergifyio update

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 75908a6cbb9266fb56260018f8f4973127c54e4a
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=e6b7fd86-fe29-4760-9f31-af3151f0fd54%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

agusti-t commented:
@Naumel thanks for merging, LMK if I can do anything from my side. I do have some unanswered outstanding questions in [this](https://github.com/aws/aws-cdk/pull/22943#issuecomment-1326791311) comment. Can you help?

Naumel commented:
https://github.com/aws/aws-cdk/pull/22943#issuecomment-1342855326

Hello,

I tried to unblock the merge from `main` and I think I made it more complicated instead of helpful.
I'll get the integ test back in shape, once the build is up and running, **all** content will be checked.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 6f3c04f66f10cfc983d403552b79d64263479b99
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=915f5c1d-0abb-4705-92f1-2cb80efa42cb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

agusti-t commented:
@rix0rrr should I try to merge or will you handle that? What about [my questions](https://github.com/aws/aws-cdk/pull/22943#issuecomment-1326791311)? Are they relevant?

agusti-t commented:
@rix0rrr and @Naumel FYI, I just rebased latest from master and force pushed, had a conflict. I didn't re run the integration test however.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 186a0f73d3112a6f3be3d5a11f578f2d4fce80be
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=aed09923-c1e0-41df-9999-e936bba5a14e%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

Naumel commented:
This PR has too many cooks, I'm bowing out.

```
@aws-cdk/aws-appsync: Snapshot Results: 
@aws-cdk/aws-appsync: Tests:    5 failed, 9 total
@aws-cdk/aws-appsync: Error: Some tests failed!
```

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
This PR cannot be merged because it has conflicts. Please resolve them. The PR will be considered stale and closed if it remains in an unmergeable state.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 9c2c7ee55990f6619ce4edd50ba1179eecb02e66
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=30dc9354-2984-442b-9fea-d4f193faaf9b%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
<!--STALE PR-->
This PR has been in the BUILD FAILING state for 3 weeks, and looks abandoned. To keep this PR from being closed, please continue work on it. If not, it will automatically be closed in a week.

TheRealAmazonKendra commented:
@Mergifyio update

TheRealAmazonKendra commented:
> @TheRealAmazonKendra Trying again, it took me a while to be able to run the tests with `yarn integ --update-on-failed`, it kept complaining about not finding the `integ` cmd, TBH I don't know how/why it started working. I had to keep trying with `yarn integ-runner --update-on-failed` and that always failed. It could also be that I rebased `main` and got the bug (suspect [this](https://github.com/aws/aws-cdk/commit/0bfce8965ee50ab79054e6f5a4c6bbecb0955e19) one) fix in, don't understand why I didn't get it earlier.
> 
> Anyway, it worked on my machine now and I committed everything, but I ask you to please check commits ([1](https://github.com/aws/aws-cdk/pull/22943/commits/544911ce7726f4a640382bbd0024ac09b9082cae), [2](https://github.com/aws/aws-cdk/pull/22943/commits/56ec81f4878d86f234cca0f122e4824faa2fde2a)) because I am not sure if these files should be committed. Of course the rest of the commits should be checked too.

On your first question, were you potentially trying to run the integ command from the package root instead of from inside the opensearch module? That's the only thing that I can think of that would cause that. Otherwise, I'm kind of stumped.

On why you didn't get the error earlier, my best guess is that when you cloned the package, the error wasn't in it yet and then upon rebasing from main, the buggy code had been merged at that point.

All speculation on my part but that's about all I can do ¯\_(ツ)_/¯ .

If the tests are still failing at this point, I can help correct them. Let's see what this build does.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 731092fc5dd169c67a6a253b0a1706016d420d63
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8ad971c6-a9e1-44d1-9c80-5e5b29c9d425%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

dblock commented:
@agusti-t can you please close this PR so we can get https://github.com/aws/aws-cdk/pull/23621#issuecomment-1379684041 merged? Thanks

dblock commented:
@TheRealAmazonKendra with Mergifyio you just updated a PR that was abandoned for 3 weeks - will we have to wait another 3 week for the bot to close it?

agusti-t commented:
> @agusti-t can you please close this PR so we can get [#23621 (comment)](https://github.com/aws/aws-cdk/pull/23621#issuecomment-1379684041) merged? Thanks

Well technically it wasn't abandoned. I was just waiting for answers to my questions in https://github.com/aws/aws-cdk/pull/22943#issuecomment-1326791311.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 255cd24b78888394b7c414030eb2df8f77185906
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=5d08b594-812f-48a4-99ec-ae04c4fef740%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

agusti-t commented:
> If the tests are still failing at this point, I can help correct them. Let's see what this build does.

@TheRealAmazonKendra fails, could you help please?

dblock commented:
@agusti-t changes here are way bigger than just adding support for OpenSearch 2.3. Can we maybe please split the changes? Feel free to do it yourself - the minimum is https://github.com/aws/aws-cdk/pull/23621 which just adds the version, and it passes tests. Then you can try making all the other updates you have here on top? It’s all yours now.

agusti-t commented:
> @agusti-t changes here are way bigger than just adding support for OpenSearch 2.3. The app sync changes are not needed for that. Can we maybe please split the changes? Feel free to do it yourself - the minimum is #23621 which just adds the version, and it passes tests. Then you can try making all the other updates you have here on top? It’s all yours now - I would close this PR, open one that looks just like 23621 and then try to make app sync updates separately.

I'm all for that, having support for 2.3 I mean. Let's see if I get help with my questions in a couple of days, otherwise I'll close and this and copy your PR (or I can let you know and you can open it yourself).

dblock commented:
I believe tests are failing because you're making changes that would be destructive during an upgrade. I'm not sure what these other changes in appsync do, but they are not related to OpenSearch 2.3 update which was the purpose of this PR.

```
@aws-cdk/aws-appsync:     Stack: aws-appsync-integ - Resource: ApiorderDsQuerygetCustomerOrdersEqResolverEF9D5350 - Impact: WILL_DESTROY
@aws-cdk/aws-appsync:     Stack: aws-appsync-integ - Resource: ApipaymentDsQuerygetPaymentResolverD172BFC9 - Impact: WILL_DESTROY
```

TheRealAmazonKendra commented:
> @TheRealAmazonKendra with Mergifyio you just updated a PR that was abandoned for 3 weeks - will we have to wait another 3 week for the bot to close it?

Four weeks, technically, but this PR was not abandoned. It was awaiting answers from our team. The appsync tests are irrelevant to this PR altogether. They got pulled in here as a result of a bug that we had in our codebase when the PR was opened that was causing the integ tests to fail. I will clean up this part. 

Once I've done so, I'll update with a new review.

TheRealAmazonKendra commented:
I think the only thing that's going to be missing from this PR is unit tests.

TheRealAmazonKendra reviewed (dismissed):
Assuming that this build passes, I just need you to add some unit tests to cover the new version and I can approve this.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: b9e07110db4198d297f4446d640fd89be839ad9e
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=7f2c589d-0aa6-4c5c-886b-0abd0e4b19c9%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: a9fca10ea308612055d6e1b943701eebee85ef4e
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=f9966e70-a59a-48f3-a81f-4a3477aee096%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

agusti-t commented:
> Contributor

I just bumped the `defaultVersion` to 2.3 and adapted existing tests. Don't know if that is enough, but if not, LMK what you'd like to see in terms of unit tests and I'll do my best. I'm a noob when it comes to ts.

dblock commented:
Thank you @agusti-t !

agusti-t commented on packages/@aws-cdk/aws-opensearchservice/test/domain.test.ts:
```diff
@@ -29,7 +29,7 @@ const readWriteActions = [
 ];
 
-const defaultVersion = EngineVersion.OPENSEARCH_1_0;
+const defaultVersion = EngineVersion.OPENSEARCH_2_3;
```
Working on it, will request review when ready.

agusti-t replied:
Done, I think it's better now. LMK what you think.

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 688065b34ecd4ca70f8dd4632f381e52e9d46a4a
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=166bb5ce-65b4-4a30-b83e-a99de44687da%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: ec7ebef54024da19e67a3270e7c10d66adc82490
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8166d964-e9f5-4e19-a23f-ef97bfaccf75%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d1280b14d8e1062fb68cab2131bbed83b4ad18c7
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ffc24033-59ef-4b05-8f31-a8c29a0849d3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: d5fdbabf9c7d0131fd02b9869f0d50fe29fa3240
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=279dba12-c20b-4c6e-8a91-056cd35294fd%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 62a6fd59fe6c0e9aac4f68ccf456df9985e23fd9
* Result: FAILED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=3c83fd4e-e92a-493b-9fdb-0d8929f0f244%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 7bfcec26c4ef9e3a3d9265cbdfc18cb43b81f777
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8f44f47e-36b9-48a9-b091-fb47d3e27615%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: af731e45afc2e465c2858d3ca185f525a5303aa8
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=8c82760d-75be-43e9-ae47-3fcb7ae813e3%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: 0f70273871868b8faa25a1400e35669de5412f88
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=d00092d9-8041-4ce3-8d2a-dafab9f37feb%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->

TheRealAmazonKendra commented on packages/@aws-cdk/aws-opensearchservice/test/domain.test.ts:
```diff
@@ -183,8 +190,14 @@ test('grants kms permissions if needed', () => {
 
-test('minimal example renders correctly', () => {
-  new Domain(stack, 'Domain', { version: defaultVersion });
+each([
```
Fantastic, thank you!

aws-cdk-automation commented:
### AWS CodeBuild CI Report

* CodeBuild project: AutoBuildv2Project1C6BFA3F-wQm2hXv2jqQv
* Commit ID: efe6e6353a41230342080f0d54c3a626313db3b2
* Result: SUCCEEDED
* [Build Logs](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=ab329d5d-a747-4017-80ff-b471849b15ce%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*


<!--
CREATED BY GITHUB-CODEBUILD-LOGS
-->
